packages/preset-wind/src/variants/misc.ts

```diff
@@ -5,7 +5,7 @@ export const variantSpaceAndDivide: Variant = (matcher) => {
     return {
       matcher,
       selector: (input) => {
-        return `${input}>:not([hidden])~:not([hidden])`
+        return `${input}>:not([hidden])`
```

Comment:
Without `~`, `space-x` will set margin-left to the left-most element while `space-y` will set margin-top to the top-most element